libs/shared/src/containers/AuthRoute/index.tsx

```diff
@@ -46,24 +47,45 @@ export const AuthRoute = (routeProps: RouteProps) => {
   )
 }
 
+// Our routes are defined as simple objects.
+// The key maps to the path, and the entry is what is rendered.
+// The map is recursive, and supports basic Route props.
+export type RouteEntry = React.ComponentType<RouteComponentProps<any>>|RouteProps|RouteMap;
+export type RouteMap = {
+  [path: string]: RouteEntry
+}
 type AuthSwitchProps = {
-  auth?: Record<string, React.ComponentType>,
-  open?: Record<string, React.ComponentType>,
+  auth?: RouteMap,
+  open?: RouteMap,
   fallback?: React.ComponentType,
-  path: string,
+  path?: string,
+  // Allow manually spec'ing the location.  This is required
+  // for cross-fade transitions (we need to render both pages at once)
+  location?: Location
 };
 
-export const AuthSwitch = ({ path, auth, open, fallback }: AuthSwitchProps) => {
-  const trimmed = path.endsWith('/') ? path.slice(0, -1) : path;
+const isRouteProps = (entry: RouteEntry) : entry is RouteProps => !!(entry as RouteProps).component;
+const isComponent = (entry: RouteEntry) : entry is React.ComponentType<RouteComponentProps<any>> => !isRouteProps(entry) && !(typeof entry == 'object');
+
+function mapEntries(root: string, mapping: Object|undefined, RouteComponent: ComponentType<RouteProps>) : JSX.Element[] {
+  if (!mapping) return [];
+  const entries = Object.entries(mapping).map(([key, entry]) => {
+    if (isRouteProps(entry)) return <RouteComponent key={key} path={`${root}/${key}`} {...entry} />
+    else if (isComponent(entry)) return <RouteComponent key={key} path={`${root}/${key}`} component={entry} />
+    else return mapEntries(`${root}/${key}`, entry, RouteComponent);
+  })
+  .reduce((acc: JSX.Element[], entry) => acc.concat(entry), []);
+  return entries;
+}
+
+export const AuthSwitch = ({ path, location, auth, open, fallback }: AuthSwitchProps) => {
```

Comment:
Switch allows for recursive route definition.  This allows us to spec all routes in a single route object